src/devices/Amg88xx/Amg88xx.cs

```diff
@@ -0,0 +1,475 @@
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Device.I2c;
+using UnitsNet;
+
+namespace Iot.Device.Amg88xx
+{
+    /// <summary>
+    /// Binding for the AMG88xx family of infrared array sensors
+    /// </summary>
+    public class Amg88xx : IDisposable
+    {
+        /// <summary>
+        /// Standard device address
+        /// (AD_SELECT pin is low, c.f. reference specification, pg. 11)
+        /// </summary>
+        public const int DeafultI2cAddress = 0x68;
+
+        /// <summary>
+        /// Alternative device address
+        /// (AD_SELECT pin is high, c.f. reference specification, pg. 11)
+        /// </summary>
+        public const int AlternativeI2cAddress = 0x69;
+
+        /// <summary>
+        /// Number of sensor pixel array columns
+        /// </summary>
+        public const int Columns = 0x8;
+
+        /// <summary>
+        /// Number of sensor pixel array rows
+        /// </summary>
+        public const int Rows = 0x8;
+
+        /// <summary>
+        /// Number of bytes per pixel
+        /// </summary>
+        private const int BytesPerPixel = 2;
+
+        /// <summary>
+        /// Temperature resolution of thermistor (in degrees Celsius)
+        /// </summary>
+        private const double ThermistorTemperatureResolution = 0.0625;
+
+        private I2cDevice _i2cDevice;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Amg88xx"/> binding.
+        /// </summary>
+        public Amg88xx(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+        }
+
+        #region Infrared sensor
+
+        /// <summary>
+        /// Gets the current thermal image.
+        /// </summary>
+        /// <returns>Array of pixel temperatures [column, row]</returns>
+        public Temperature[,] GetThermalImage()
+        {
+            // the readout process gets triggered by writing to pixel 0 of the sensor w/o any additional data
+            _i2cDevice.WriteByte((byte)Register.T01L);
+
+            Span<byte> buffer = stackalloc byte[Rows * Columns * BytesPerPixel];
+            _i2cDevice.Read(buffer);
+
+            var image = new Temperature[Columns, Rows];
+
+            int idx = 0;
+            for (int r = 0; r < Rows; r++)
+            {
+                for (int c = 0; c < Columns; c++)
+                {
+                    byte tl = buffer[idx++];
+                    byte th = buffer[idx++];
+                    image[c, r] = Amg88xxUtils.ConvertToTemperature(tl, th);
+                }
+            }
+
+            return image;
+        }
+
+        /// <summary>
+        /// Gets the current raw thermal image from the sensor.
+        /// </summary>
+        /// <returns>Thermal image in two's complement representation</returns>
+        public int[,] GetThermalRawImage()
```

Comment:
since you're already using Span consider allowing for users to pass the buffer in and also making Stride public

Reply:
@krwq I actually thought about this, but I rather prefer the current API. If I would expect the user to provide a Span I would have to reveal that each pixel requires 2 bytes. From my point of view this is a technical detail the user shouldn't need to know. Can you agree to this?

Reply:
@fehdem yes and no, it is a technical detail but all graphics library reveal this detail for perf reasons. You will allocate much more data the way you have it right now. Other 2 alternatives which allow you to avoid allocations on each frame and allow similar APIs:
- create a struct/class (i.e. AMG88xxImage) which has a `byte[]` internally which you can pass in here and will be filled in this method and also have indexer for that class to hide internal data format
- inside of this class have a `byte[]` which can store the whole image, have a method Read/ReadImage or similar which does not return anything (or perhaps bool if it can fail but then you call it TryRead) and have an indexer here implemented i.e.:

```csharp
public class MyClass
{
    public const int Width = 3;
    public const int Height = 5;
    private const int BytesPerPixel = 2;
    private byte[] _bytes = new byte[BytesPerPixel * Width * Height];

    private UInt16 ConvertToTemperature(ReadOnlySpan<byte> rawTemperature)
    {
        // whatever the format for reading is, consider inlining this method
        // if this is exactly as below then consider using ReadOnlySpan<UInt16> directly
        // in the previous method and not use this at all
        return BinaryPrimitives.ReadUInt16LittleEndian(rawTemperature);
    }

    public void Read()
    {
        // Fill the _bytes here, something along the lines of:
        // _i2cDevice.Read(_bytes);
    }

    public UInt16 this[int x, int y]
    {
        get
        {
            if (x < 0 || x >= Width)
                throw new ArgumentOutOfRangeException(nameof(x));
            if (y < 0 || y >= Height)
                throw new ArgumentOutOfRangeException(nameof(y));

            // fix the byte mapping to whatever the actual is
            Span<byte> buffer = _bytes;
            Span<byte> temperature = buffer.Slice(BytesPerPixel * (Width * y + x), BytesPerPixel);
            return ConvertToTemperature(temperature);
        }
    }
}
```

Reply:
@krwq I got your point of reducing the (re-)allocations of memory for each frame.
I think about another adjustment to the API out of this.
I like the indexer approach you suggested. But, I have the temperature-converted image and the raw image. I want to keep both options, because both have reasonable use cases (I use the temperatures for display purpose and I feed the raw values into a neural network.) 
However, I think I could make the raw image even more raw. Instead of using an x/y-index I could use the pixel number (0..63) only. This would meet even better the neuronal network use case. 
So I would end up with two indexers:
```
public Temperature this[int x, int y]
public UInt16 this[int pixel]
```
What do you think about this idea?

Reply:
> What do you think about this idea?

I personally like the idea! Makes sense

Reply:
@fehdem sounds good but to improve perf you can also make pixel array user provided, i.e.:

```csharp
public void ReadPixels(Span<UInt16> pixels)
```

Reply:
I think the API is good with the indexers and the internally kept pixel buffer.